lib/core/test/unit/Cardano/Wallet/ApiSpec.hs

```diff
@@ -169,41 +173,40 @@ spec = parallel $ do
             forM_ tests $ \(req, msg) -> it (titleize proxy req) $
                 runSession (spec_NotAllowedMethod req msg) application
 
-spec_MalformedParam :: Request -> ExpectedError -> Session ()
-spec_MalformedParam malformedRequest (ExpectedError msg) = do
-    response <- request malformedRequest
-    response & assertStatus 400
+assertErrorResponse
+    :: HasCallStack
+    => Int -- ^ Expected status
+    -> Text -- ^ Expected error code string
+    -> ExpectedError
+    -> SResponse
+    -> Session ()
+assertErrorResponse status code (ExpectedError msg) response = do
+    response & assertStatus status
+    response & assertHeader "Content-Type" "application/json;charset=utf-8"
```

Comment:
:clap:

Reply:
These tests don't actually cover the `no_such_wallet` error reported in issue #2596.

Reply:
Yes indeed, the `404` are sort of cumbersome to trigger with this generic approach which only focuses on what exists.